experimental/devices/mfrc522/mfrc522.go

```diff
@@ -0,0 +1,767 @@
+		return
+	}
+	err = r.devWrite(address, current|byte(mask))
+	return
+}
+
+func (r *Dev) clearBitmask(address, mask int) (err error) {
+	current, err := r.devRead(address)
+	if err != nil {
+		return
+	}
+	err = r.devWrite(address, current&^byte(mask))
+	return
+
+}
+
+// Reset resets the RFID chip to initial state.
+func (r *Dev) Reset() (err error) {
+	r.Authenticated = false
+	err = r.devWrite(commands.CommandReg, commands.PCD_RESETPHASE)
+	return
+}
+
+// SetAntenna configures the antenna state, on/off.
+func (r *Dev) SetAntenna(state bool) (err error) {
+	if state {
+		current, err := r.devRead(commands.TxControlReg)
+		if err != nil {
+			return err
+		}
+		if current&0x03 == 0 {
+			err = r.setBitmask(commands.TxControlReg, 0x03)
+		}
+	} else {
+		err = r.clearBitmask(commands.TxControlReg, 0x03)
+	}
+	return
+}
+
+/*
+CardWrite the low-level interface to write some raw commands to the card.
+	command - the command register
+	data - the data to write out to the card using the authenticated sector.
+*/
+func (r *Dev) CardWrite(command byte, data []byte) (backData []byte, backLength int, err error) {
+	backLength = -1
+	irqEn := byte(0x00)
+	irqWait := byte(0x00)
+
+	switch command {
+	case commands.PCD_AUTHENT:
+		irqEn = 0x12
+		irqWait = 0x10
+	case commands.PCD_TRANSCEIVE:
+		irqEn = 0x77
+		irqWait = 0x30
+	}
+
+	r.devWrite(commands.CommIEnReg, irqEn|0x80)
+	r.clearBitmask(commands.CommIrqReg, 0x80)
+	r.setBitmask(commands.FIFOLevelReg, 0x80)
+	r.devWrite(commands.CommandReg, commands.PCD_IDLE)
+
+	for _, v := range data {
+		r.devWrite(commands.FIFODataReg, v)
+	}
+
+	r.devWrite(commands.CommandReg, command)
+
+	if command == commands.PCD_TRANSCEIVE {
+		r.setBitmask(commands.BitFramingReg, 0x80)
+	}
+
+	i := 2000
+	n := byte(0)
+
+	for ; i > 0; i-- {
+		n, err = r.devRead(commands.CommIrqReg)
+		if err != nil {
+			return
+		}
+		if n&(irqWait|1) != 0 {
+			break
+		}
+	}
+
+	r.clearBitmask(commands.BitFramingReg, 0x80)
+
+	if i == 0 {
+		err = errors.New("can't read data after 2000 loops")
+		return
+	}
+
+	if d, err1 := r.devRead(commands.ErrorReg); err1 != nil || d&0x1B != 0 {
+		err = err1
+		return
+	}
+
+	if n&irqEn&0x01 == 1 {
+		err = errors.New("IRQ error")
+		return
+	}
+
+	if command == commands.PCD_TRANSCEIVE {
+		n, err = r.devRead(commands.FIFOLevelReg)
+		if err != nil {
+			return
+		}
+		lastBits, err1 := r.devRead(commands.ControlReg)
+		if err1 != nil {
+			err = err1
+			return
+		}
+		lastBits = lastBits & 0x07
+		if lastBits != 0 {
+			backLength = (int(n)-1)*8 + int(lastBits)
+		} else {
+			backLength = int(n) * 8
+		}
+
+		if n == 0 {
+			n = 1
+		}
+
+		if n > 16 {
+			n = 16
+		}
+
+		backData = make([]byte, n)
+		for i := byte(0); i < n; i++ {
+			byteVal, err1 := r.devRead(commands.FIFODataReg)
+			if err1 != nil {
+				err = err1
+				return
+			}
+			backData[i] = byteVal
+		}
+
+	}
+
+	return
+}
+
+// Request request the card information. Returns number of blocks available on the card.
```

Comment:
// Request requests the card information.

(yeah I know)
But still, that's kind of fuzzy what this means.